packages/nix/mkDevShells/pf-env-bootstrap-restore.sh

```diff
@@ -0,0 +1,255 @@
+
+  if [[ ! -f $STATE_FILE ]]; then
+    echo "State file not found."
+    return 1
+  fi
+
+  # Retrieve the property value using jq
+  jq -r --arg cluster "$EKS_CLUSTER_NAME" --arg property_name "$property_name" \
+    '.[$cluster][$property_name] // "Property not found."' "$STATE_FILE"
+}
+
+aws_profile=$(get_cluster_state "aws_profile")
+export AWS_PROFILE=$aws_profile
+
+if [[ -z $AWS_PROFILE ]]; then
+  echo "Error: AWS profile not found in state file. Exiting." >&2
+  exit 1
+fi
+
+scale_asgs_nat_nodes() {
+  local prefix="nat-"
+  local min_size=$1
+  local desired_capacity=$2
+
+  asgs=$(aws autoscaling describe-auto-scaling-groups | jq -r ".AutoScalingGroups[] | select(.AutoScalingGroupName | startswith(\"$prefix\")) | .AutoScalingGroupName")
+
+  if [ -z "$asgs" ]; then
+    echo "No Auto Scaling Groups found that start with '$prefix'."
+    return
+  fi
+
+  for asg in $asgs; do
+    echo "Updating ASG: $asg"
+
+    if aws autoscaling update-auto-scaling-group --auto-scaling-group-name "$asg" --min-size "$min_size" --desired-capacity "$desired_capacity"; then
+      echo "Successfully updated $asg to min-size $min_size and desired-capacity $desired_capacity."
+    else
+      echo "Failed to update $asg."
+    fi
+  done
+}
+
+scale_asg_controller_nodes() {
+  local min_size=$1
+  local desired_capacity=$2
+  local nodegroup_prefix="controllers-"
+
+  echo "$min_size |  $desired_capacity"
+
+  asgs=$(aws autoscaling describe-auto-scaling-groups | jq -r ".AutoScalingGroups[] | select(
+             (.Tags[] | select(.Key == \"eks:cluster-name\" and .Value == \"$EKS_CLUSTER_NAME\")) and
+             (.Tags[] | select(.Key == \"eks:nodegroup-name\" and (.Value | startswith(\"$nodegroup_prefix\"))))
+           ) | .AutoScalingGroupName")
+
+  if [ -z "$asgs" ]; then
+    echo "No Auto Scaling Groups found for EKS cluster '$EKS_CLUSTER_NAME' with nodegroup starting with '$nodegroup_prefix'."
+    return
+  fi
+
+  for asg in $asgs; do
+    echo "$asg" --min-size "$min_size" --desired-capacity "$desired_capacity"
+    echo "Updating ASG: $asg"
+
+    if aws autoscaling update-auto-scaling-group --auto-scaling-group-name "$asg" --min-size "$min_size" --desired-capacity "$desired_capacity"; then
+      echo "Successfully updated $asg to min-size $min_size and desired-capacity $desired_capacity."
+    else
+      echo "Failed to update $asg."
+    fi
+  done
+}
+
+update_scheduler() {
+  local namespace=$1
+  local deployment_name=$2
+  local scheduler_name=$3
+
+  if kubectl "$CONTEXT_ARGS" patch deployment "$deployment_name" -n "$namespace" --patch "{
+           \"spec\": {
+             \"template\": {
+               \"spec\": {
+                 \"schedulerName\": \"$scheduler_name\"
+               }
+             }
+           }
+  }"; then
+    echo "Successfully updated the scheduler to $scheduler_name for $deployment_name in namespace $namespace."
+  else
+    echo "Failed to update the scheduler for $deployment_name."
```

Comment:
This can fail if the deployment does not exist (cilium not installed yet). This message seems to imply the script failed but I don't think that is a script failure case.